### PR TITLE
Mayhem

### PR DESCRIPTION
**Pull request by capuanob**
This includes my contributions for harnessing MLE with Mayhem. Please let me know if anything needs to be adjusted!

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 prefix?=/usr/local
 
-mle_cflags:=-std=c99 -Wall -Wextra -pedantic -Wno-pointer-arith -Wno-unused-result -Wno-unused-parameter -g -O0 -D_GNU_SOURCE -I. $(CFLAGS)
```
Comment (xansec) [inline Makefile:3]: You should be able to pass `-fsanitize=fuzzer` in the `CFLAGS` environment variable without modifying the Makefile.

**Review by xansec — CHANGES_REQUESTED** [review]
Please do not modify source unless necessary (i.e. build is broken). I don't think upstream will accept your changes as is. I would create maybe a new module or `fuzz` folder, and create a fuzz target in there rather than modifying existing source.

**Comment by capuanob** [comment]
Did a total rework on fuzzing harness to be 1) More accurate to how MLE would truly be used and 2) Prevent making source code changes

**Comment by xansec** [comment]
Great improvement!

**Pull request merged**